examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -442,20 +557,94 @@ class DreamBoothDataset(Dataset):
     def __init__(
         self,
         instance_data_root,
+        instance_prompt,
+        class_prompt,
         class_data_root=None,
         class_num=None,
         size=1024,
+        repeats=1,
         center_crop=False,
     ):
         self.size = size
         self.center_crop = center_crop
 
         self.instance_data_root = Path(instance_data_root)
-        if not self.instance_data_root.exists():
-            raise ValueError("Instance images root doesn't exists.")
 
-        self.instance_images_path = list(Path(instance_data_root).iterdir())
-        self.num_instance_images = len(self.instance_images_path)
+        self.instance_prompt = instance_prompt
+        self.custom_instance_prompts = None
+        self.class_prompt = class_prompt
+
+        load_as_dataset = False
+
+        # if --dataset_name is provided or a metadata jsonl file is provided in the local --instance_data directory,
+        # we load the training data using load_dataset
+        if args.dataset_name is not None:
+            from datasets import load_dataset
+
+            # Downloading and loading a dataset from the hub.
+            # See more about loading custom images at
+            # https://huggingface.co/docs/datasets/v2.0.0/en/dataset_script
+            dataset = load_dataset(
+                args.dataset_name,
+                args.dataset_config_name,
+                cache_dir=args.cache_dir,
+            )
+            load_as_dataset = True
+        else:
+            if not self.instance_data_root.exists():
```

Comment:
in this `else:` case we shouldn't use `datasets`. Can we make sure that images are loaded without using datasets?

Reply:
E.g. using the logic from before the change: 
```py
        if not self.instance_data_root.exists():
            raise ValueError("Instance images root doesn't exists.")

        self.instance_images_path = list(Path(instance_data_root).iterdir())
        self.num_instance_images = len(self.instance_images_path)
```

Reply:
This else refers to the case where the user has locally saved prompts - a local folder with a jsonl file, i.e:
- user provides `--instance_data_dir` and `--instance_data_metadata_file_name`
(I did so to allow custom captions for local folders)

The original logic - loading without dataset - is still applied for local folders with images only. i.e:

-  when the user provides only `--instance_data_dir` and not `--instance_data_metadata_file_name`